File: Preparation Kit/Week 1/1_Plus_Minus.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'plusMinus' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static void plusMinus(List<Integer> arr) {
    // Write your code here
        int n = arr.size();
        int positives = 0;
        int negatives = 0;
        int zeros = 0;

        for (int num : arr) {
            if (num > 0) {
                positives++;
            } else if (num < 0) {
                negatives++;
            } else {
                zeros++;
            }
        }

        double positive_ratio = (double) positives / n;
        double negative_ratio = (double) negatives / n;
        double zero_ratio = (double) zeros / n;

        System.out.printf("%.6f%n", positive_ratio);
        System.out.printf("%.6f%n", negative_ratio);
        System.out.printf("%.6f%n", zero_ratio);
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        Result.plusMinus(arr);

        bufferedReader.close();
    }
}

File: Preparation Kit/Week 1/1_Plus_Minus.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def plusMinus(arr):
    # Write your code here
    n = len(arr)
    positives = sum(1 for x in arr if x > 0)
    negatives = sum(1 for x in arr if x < 0)
    zeros = sum(1 for x in arr if x == 0)

    positive_ratio = positives / n
    negative_ratio = negatives / n
    zero_ratio = zeros / n

    print(f"{positive_ratio:.6f}")
    print(f"{negative_ratio:.6f}")
    print(f"{zero_ratio:.6f}")

if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)
